modules/tinymce/src/plugins/link/test/ts/browser/ImageFigureLinkTest.ts

```diff
@@ -1,59 +1,42 @@
-import { UnitTest } from '@ephox/bedrock-client';
-import { TinyApis, TinyDom, TinyLoader, TinyUi } from '@ephox/mcagar';
+import { describe, it, before, after } from '@ephox/bedrock-client';
+import { TinyHooks, TinySelections } from '@ephox/mcagar';
+import Editor from 'tinymce/core/api/Editor';
 import * as LinkPluginUtils from 'tinymce/plugins/link/core/Utils';
-import LinkPlugin from 'tinymce/plugins/link/Plugin';
-import SilverTheme from 'tinymce/themes/silver/Theme';
-import { TestLinkUi } from '../module/TestLinkUi';
-
-UnitTest.asynctest('browser.tinymce.plugins.link.ImageFigureLinkTest', (success, failure) => {
-
-  LinkPlugin();
-  SilverTheme();
-
-  TinyLoader.setupLight((editor, onSuccess, onFailure) => {
-    const api = TinyApis(editor);
-    const ui = TinyUi(editor);
-
-    const sLinkTheSelection = () => {
-      return Logger.t('Link the selection', TestLinkUi.sInsertLink(ui, 'http://google.com'));
-    };
+import Plugin from 'tinymce/plugins/link/Plugin';
+import Theme from 'tinymce/themes/silver/Theme';
 
-    const sUnlinkSelection = () => {
-      return Logger.t('Unlink the selection', Step.sync(() => {
-        LinkPluginUtils.unlink(editor);
-      }));
-    };
-
-    const sAssertPresence = (selector: Record<string, number>) => {
-      return Waiter.sTryUntil('Assert element is present',
-        Assertions.sAssertPresence('Detect presence of the element', selector, TinyDom.fromDom(editor.getBody()))
-      );
-    };
-
-    Pipeline.async({},
-      Log.steps('TBA', 'Link: Set content, select and link the selection, assert link is present. Then select and unlink the selection, assert link is not present', [
-        TestLinkUi.sClearHistory,
-        api.sSetContent(
-          '<figure class="image">' +
-            '<img src="http://moxiecode.cachefly.net/tinymce/v9/images/logo.png" />' +
-            '<figcaption>TinyMCE</figcaption>' +
-          '</figure>'
-        ),
-        api.sSetSelection([ 0 ], 0, [ 0 ], 0),
-        sLinkTheSelection(),
-        sAssertPresence({ 'figure.image > a[href="http://google.com"] > img': 1 }),
+import { TestLinkUi } from '../module/TestLinkUi';
 
-        api.sSetSelection([ 0 ], 0, [ 0 ], 0),
-        sUnlinkSelection(),
-        sAssertPresence({ 'figure.image > img': 1 }),
-        TestLinkUi.sClearHistory
-      ])
-      , onSuccess, onFailure);
-  }, {
+describe('browser.tinymce.plugins.link.ImageFigureLinkTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
     plugins: 'link',
     toolbar: 'link',
-    theme: 'silver',
     base_url: '/project/tinymce/js/tinymce'
-  }, success, failure);
+  }, [ Plugin, Theme ]);
+
+  before(() => {
+    TestLinkUi.clearHistory();
+  });
+
+  after(() => {
+    TestLinkUi.clearHistory();
+  });
+
+  it('TBA: Set content, select and link the selection, assert link is present. Then select and unlink the selection, assert link is not present', async () => {
```

Comment:
Can this be broken up into 2 tests? It seems to be testing 2 things, first inserting a link and then secondly removing a link.

Reply:
Yeah, it should be two tests. I have split it up.